services/httpd/service.go

```diff
@@ -192,6 +197,23 @@ func (s *Service) Open() error {
 func (s *Service) Close() error {
 	s.Handler.Close()
 
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	s.httpServerMutex.Lock()
+	errCh := make(chan error, len(s.httpServer))
+	// shut down all running http servers
+	for _, server := range s.httpServer {
```

Comment:
Should the two `for` loops  be a go func, with the `unlock` in a `defer` so that if the context timeout occurs, you don't hold the mutex?  Maybe it's cleared automatically; I'm not certain how Go does it.  So, lines 206 - 214 become a synchronously called go func (which spawns the asynchronous shutdowns as currently), but has line 215 as a defer on that little func?